File: 2073. Time Needed to Buy Tickets.py
#* 2073. Time Needed to Buy Tickets
tickets = [2,3,2]; k = 2
#* MY SOL 
def time_required_to_buy(tickets, k):
    seconds = 0
    for i, ticket in enumerate(tickets):
        seconds += min(ticket, tickets[k])
        if i > k and ticket >= tickets[k]:
            seconds -= 1
    return seconds

print(time_required_to_buy(tickets, k))

#*
def time_required_to_buy(tickets, k):
    return sum(
        min(ticket, tickets[k]) - (i > k and ticket >= tickets[k])
        for i, ticket in enumerate(tickets)
    )

print(time_required_to_buy(tickets, k))

#* compact
def time_required_to_buy(tickets, k):
    return sum(min(ticket, tickets[k]) - (i > k and ticket >= tickets[k]) for i, ticket in enumerate(tickets))

print(time_required_to_buy(tickets, k))

File: 2073. Time Needed to Buy Tickets.js
// 2073. Time Needed to Buy Tickets
tickets = [2,3,2]; k = 2
// MY SOL 
/**
 * @param {number[]} tickets
 * @param {number} k
 * @return {number}
 */
var timeRequiredToBuy = function(tickets, k) {
  return tickets.reduce((seconds, ticket, i) => {
    return seconds + Math.min(ticket, tickets[k]) - (i > k && ticket >= tickets[k] ? 1 : 0);
  }, 0);
}
console.log(timeRequiredToBuy(tickets, k));